src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -799,6 +800,36 @@ private static bool CanReplace(ISymbol symbol)
                 }
             }
 
+            // Try to eliminate cases without actually calling CanReplaceWithReducedName. For expressions of the form
+            // 'this.Name' or 'base.Name', no additional check here is required.
+            if (!memberAccess.Expression.IsKind(SyntaxKind.ThisExpression, SyntaxKind.BaseExpression))
+            {
+                var actualSymbol = semanticModel.GetSymbolInfo(memberAccess.Name, cancellationToken);
+                if (actualSymbol.GetAnySymbol() is null)
+                {
+                    return false;
+                }
+
+                var symbols = semanticModel.LookupSymbols(memberAccess.SpanStart, name: memberAccess.Name.Identifier.ValueText);
```

Comment:
:bulb: Technically we can avoid the call to `LookupSymbols` and go straight for `LookupNamespacesAndTypes` any time `actualSymbol` only contains namespace and/or type symbols. This will actually be a common case for Simplify Type Name.